pydantic/generics.py

```diff
@@ -56,6 +56,8 @@ def __class_getitem__(  # type: ignore
         created_model.Config = cls.Config
         created_model.__concrete = True  # type: ignore
         _generic_types_cache[(cls, params)] = created_model
+        if len(params) == 1:
+            _generic_types_cache[(cls, params[0])] = created_model
         return created_model
```

Comment:
You need to have the non-tuple version in the cache so cache-access can be the first line of the function (rather than tuple-making), ensuring minimum overhead. I think it makes sense to also add the tuple version to the cache to minimize the amount of special cases (and after all it is possible to pass a single-element tuple to the `__class_getitem__` function).